core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1607,11 +1616,18 @@ private[spark] class DAGScheduler(
                   case mapStage: ShuffleMapStage =>
                     val numMissingPartitions = mapStage.findMissingPartitions().length
                     if (numMissingPartitions < mapStage.numTasks) {
-                      // TODO: support to rollback shuffle files.
-                      // Currently the shuffle writing is "first write wins", so we can't re-run a
-                      // shuffle map stage and overwrite existing shuffle files. We have to finish
-                      // SPARK-8029 first.
-                      abortStage(mapStage, generateErrorMessage(mapStage), None)
+                      if (sc.getConf.get(config.SHUFFLE_USE_OLD_FETCH_PROTOCOL)) {
```

Comment:
For the compatibility with the scenario of old fetch protocol(e.g. external shuffle service can not upgrade now), we only do the extension in new version shuffle fetch protocol, which introduced in #24565.

Reply:
what about general shuffles? It looks to me that the shuffle reader needs to keep both the legacy int map id and the new long map id. When `SHUFFLE_USE_OLD_FETCH_PROTOCOL`, construct the new shuffle request with the new long map id. Otherwise, construct the old shuffle request with the legacy int map id.

Reply:
Thanks for the reminding, I missed the protocol checking in MapOutputTracker, fix this in 
539d725.